### ISSUE TITLE
Crash when opening project with autoload singleton in 3.2 RC3 mono

### ISSUE THREAD

**Issue by Torguen** [godotengine/godot#35547]
**Godot version:**
3.2 RC 3 mono

**OS/device including version:**
win10 64 home

**Issue description:**
A crash occurs when opening the attached project.
Deleting the .mono folder does not solve anything.

**Steps to reproduce:**
Open the attached project

**Minimal reproduction project:**
[crashProject.zip](https://github.com/godotengine/godot/files/4111803/crashProject.zip)

**Comment by akien-mga** [comment]
Confirmed on Linux with 3.2 RC 3 Mono. Clearing the `.mono` folder indeed does not fix it.

**Comment by akien-mga** [comment]
Also reproducible in the current master branch (f6781411), `gdb` backtrace:
```
ERROR: _create_project_solution_if_needed: FATAL: Condition ' CSharpLanguage::get_singleton()->get_godotsharp_editor() == __null ' is true.
   At: modules/mono/csharp_script.cpp:75.

Thread 1 "godot-mono" received signal SIGILL, Illegal instruction.

(gdb) bt
#0  0x0000000001a2f5f6 in _create_project_solution_if_needed () at modules/mono/csharp_script.cpp:75
#1  0x0000000001a3c611 in CSharpScript::can_instance (this=0x72095a0) at modules/mono/csharp_script.cpp:2889
#2  0x0000000003991dca in Object::set_script (this=0x6b2f5a0, p_script=...) at core/object.cpp:996
#3  0x000000000398eda7 in Object::set (this=0x6b2f5a0, p_name=..., p_value=..., r_valid=0x7fffffff7daf) at core/object.cpp:432
#4  0x00000000033779ff in SceneState::instance (this=0x9209da0, p_edit_state=SceneState::GEN_EDIT_STATE_DISABLED) at scene/resources/packed_scene.cpp:209
#5  0x0000000003382684 in PackedScene::instance (this=0x885b460, p_edit_state=PackedScene::GEN_EDIT_STATE_DISABLED) at scene/resources/packed_scene.cpp:1695
#6  0x0000000002a76d1b in EditorAutoloadSettings::_create_autoload (this=0x8854f90, p_path=...) at editor/editor_autoload_settings.cpp:324
#7  0x0000000002a7c065 in EditorAutoloadSettings::EditorAutoloadSettings (this=0x8854f90) at editor/editor_autoload_settings.cpp:773
#8  0x0000000002454b3e in ProjectSettingsEditor::ProjectSettingsEditor (this=0x86a16b0, p_data=0x6d4e5f0) at editor/project_settings_editor.cpp:2119
#9  0x00000000022b0897 in EditorNode::EditorNode (this=0x6d4de90) at editor/editor_node.cpp:6086
#10 0x00000000014cf28b in Main::start () at main/main.cpp:1726
#11 0x0000000001492f39 in main (argc=2, argv=0x7fffffffd738) at platform/x11/godot_x11.cpp:55
```

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by Torguen** [comment]
It worked fine on my 3.2.4 beta 4 mono

**Comment by akien-mga** [comment]
Thanks, assuming fixed in a 3.2.x release.
